feat(docker): Build multiple python version images

Germandrummer92 opened the pull request:
### SUMMARY
We would like to use python3.9/3.10 in our custom configuration code to have some nice-to-have features. I don't really see a reason why there shouldn't be a python3.9/python3.10 version, the build arg even already exists?

### TESTING INSTRUCTIONS
Run the docker build and see it works with python3.9/python3.10

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [X] Introduces new feature or API
- [ ] Removes existing feature or API

Commit:
build multiple python version images

rusackas commented:
Seems related to https://github.com/apache/superset/discussions/22810

craig-rueda commented on .github/workflows/docker_build_push.sh:
```diff
@@ -50,12 +50,39 @@ docker build --target lean \
+  --build-arg PY_VER="3.9.16"\
+  --label "sha=${SHA}" \
+  --label "built_at=$(date)" \
+  --label "target=lean39" \
```
I don't think this target exists in the `Dockerfile`

craig-rueda commented on .github/workflows/docker_build_push.sh:
```diff
@@ -50,12 +50,39 @@ docker build --target lean \
+
+
+# Build the "lean310" image
+docker build --target lean \
```
Let's only go with `3.9` for now as `3.10` isn't tested yet

Germandrummer92 replied:
removed 3.10

Commit:
MR review

Commit:
slim versions

Germandrummer92 commented:
@rusackas I removed the patch version now and switched back to the -slim version. Let me know if that makes sense or if we should stay with the pinned patch version. 

I think -slim for sure makes sense if it works, which it did before.

Germandrummer92 commented:
Maybe there also should be a defined process of when the "next-python" image becomes the "main" python version? 

python3.8 is still supported until next year, but in general I wouldn't mind if we only lag about 1 year behind the python release schedule? https://endoflife.date/python

That would mean we would try to move the "main" python image to 3.9 and have 3.10 support soon; and move to 3.10/3.11 in October when 3.12 releases?
